scylla.go

```diff
@@ -107,7 +107,12 @@ func newScyllaConnPicker(conn *Conn) *scyllaConnPicker {
 func (p *scyllaConnPicker) Remove(conn *Conn) {
 	s := parseSupported(conn.supported)
 	if s.nrShards == 0 {
-		panic(fmt.Sprintf("scylla: %s not a sharded connection", conn.Address()))
+		// It is possible for Remove to be called before the connection is added to the pool.
+		// Ignoring these connections here is safe.
+		if gocqlDebug {
+			Logger.Printf("scylla: %s not a sharded connection", conn.Address())
```

Comment:
how about `scylla: %s remove alien connection`?

Reply:
As appealing as that seems does it make it clearer? How about `scylla: %s has unknown sharding state, ignoring it`